exercises/src/main/scala/fpinscala/iomonad/IO.scala

```diff
@@ -306,18 +308,41 @@ object IO3 {
                                f: A => Free[F, B]) extends Free[F, B]
 
   // Exercise 1: Implement the free monad
-  def freeMonad[F[_]]: Monad[({type f[a] = Free[F,a]})#f] = ???
+  def freeMonad[F[_]]: Monad[({type f[a] = Free[F,a]})#f] = new Monad[({type f[a] = iomonad.Free[F, a]})#f] {
```

Comment:
You should take a look at https://github.com/non/kind-projector it really saves you this boilerplate.